### DISCUSSION TITLE
Quill in Livewire 3 - wire:ignore issue

### DISCUSSION THREAD

**Discussion by instantia-data** [livewire/livewire#8391] [7. Bugs (archived)]
### Livewire version

3.4.10

### Laravel version

11.4.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I'm trying to implement the text editor within a form developed in Livewire 3.
When I submit the form the editor simply disappears.
Basically neither the wire:ignore nor wire:ignore.self directives do not work.

<div wire:ignore>
    <div class="form-group mb-2">
        <label>Content</label>
        <div id="editor_{{ $name }}" wire:ignore></div>
        
        <input type="hidden" name="{{ $name }}" wire:model="{{$name}}">
    </div>

    @push('scripts')
        <script>
            document.addEventListener('livewire:init', () => {
                const editorId = 'editor_{{ $name }}';
                const editorElement = document.getElementById(editorId);
                const hiddenInput = document.querySelector('input[name="{{$name}}"]');

                if (!editorElement.__quill) {
                    const quill = new Quill(`#${editorId}`, {
                        theme: 'snow'
                    });
                    setTimeout(function () {
                        quill.root.innerHTML = hiddenInput.value;
                    }, 100);
                    editorElement.__quill = quill;
                    quill.on('text-change', function () {
                        hiddenInput.value = quill.root.innerHTML;
                        hiddenInput.dispatchEvent(new Event('input')); // Trigger Livewire's update manually
                    });
                }
            });
        </script>
    @endpush
</div>

### Repository or Wirebox URL

https://github.com/quilljs

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by francoism90** [comment]
Because you are on Livewire V3, it's better to rewrite your form and use the `@scripts` helper and an AlpineJS data object.

**Reply by instantia-data** [reply]
of course i use @scripts helper.
Which AlpineJS data object do you mean?

**Comment by jeffegiovani** [comment]
I also had problems with `wire:ignore`, the problem in my case was the **HTML minification** that I was doing via middleware. I don't know exactly what caused the problem, but it was probably corrupting the Livewire snapshot.
I hope it helped in some way.

**Reply by francoism90** [reply]
@jeffegiovani What do you use to compress the HTML? Did you add the middleware as persist?

**Reply by jeffegiovani** [reply]
I'm using the `abordage/laravel-html-min` package to compress. Good tip for using persistent middleware, I tried but it didn't work.
Using the `#[Lazy]` _(Livewire\Attributes\Lazy)_, was the solution I found.

**Reply by francoism90** [reply]
@jeffegiovani I don't know if this works with Livewire and JS-frameworks, because you minify the output and so it doesn't match anymore.

Other packages suggest adding some stuff to the JS-output, like semicolons.

I don't know if HTML minify makes any performance improvements nowadays.

**Comment by naknak987** [comment]
Here are the versions I'm using.
PHP 8.1
Laravel 10.10
Livewire 3.5

Wrapping my Quill div in another div with the `wire:ignore` attribute worked for me. The editing pane and toolbar persist form submissions.

Like this.
```
<div wire:ignore>
    <div id="quill_element"></div>
</div>
```

Setting `wire:ignore` on the `quill_element` itself results in the toolbar disappearing because the toolbar is placed in the dom before the `quill_element`. So, the toolbar itself would not be ignored when `wire:ignore` is set on the `quill_element` itself. Hence, the `quill_element` must be nested in an element with `wire:ignore`.

It's not clear if this is the same issue you were facing, but I hope this helps someone.

**Reply by tegarmonster** [reply]
thanks @naknak987 it's work
